### PR TITLE
[0421] Course search initial flow - validation

### PR DESCRIPTION
**Pull request by defong**
### Context

Course search initial flow - validation

### Changes proposed in this pull request

Added Course search initial flow - validation

### Guidance to review

image

image

image

Clicking `continue` when you have pass validation will error out as the user journey is in complete.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Course Search - initial flow validation](https://trello.com/c/t7wx8HB3/421-course-search-initial-flow-validation)

File: app/controllers/find/search/locations_controller.rb [hunk 1]
```diff
@@ -5,6 +5,26 @@ def index
...
         @courses_by_location_or_training_provider_form = CoursesByLocationOrTrainingProviderForm.new
       end
+
+      def create
+        @courses_by_location_or_training_provider_form = CoursesByLocationOrTrainingProviderForm.new(params: find_courses_by_location_or_training_provider_form_params)
+
+        if @courses_by_location_or_training_provider_form.valid?
+          redirect_to find_age_groups_path
+        else
+          @providers = RecruitmentCycle.current.providers.by_name_ascending
```
Comment (darokel) [inline app/controllers/find/search/locations_controller.rb:15]: this and the same query on line 5 could use:

```ruby
def providers
  @providers ||= RecruitmentCycle.current.providers.by_name_ascending
end
```

File: app/javascript/find/setup-autocomplete.js [hunk 2]
```diff
@@ -1,9 +1,14 @@
 import accessibleAutocomplete from 'accessible-autocomplete'
 
-const getSelectElement = inputId => document.getElementById(inputId)
+const getSelectElement = elementId => document.getElementById(elementId)
 
 const setupAutocomplete = inputId => {
-  const selectElement = getSelectElement(inputId)
+  let elementId = inputId
+  let selectElement = getSelectElement(elementId)
+  if (selectElement === null) {
```
Comment (darokel) [inline app/javascript/find/setup-autocomplete.js:8]:
```suggestion
if (!selectElement) {
```

**Review by darokel — APPROVED** [review]
Left two cosmetic suggestions but no blockers

File: app/controllers/find/search/locations_controller.rb (hunk 1 above)
Comment (defong) [reply app/controllers/find/search/locations_controller.rb]: fixed

File: app/javascript/find/setup-autocomplete.js (hunk 2 above)
Comment (defong) [reply app/javascript/find/setup-autocomplete.js]: fixed

**Pull request merged**